### ISSUE TITLE
pose_parameter() proxy function doesn't work, returns 0

### ISSUE THREAD

**Issue by pingu7867** [CapsAdmin/pac3#1177]
Yes the proxy is root owner. It's inconsistent, though. Earlier this evening it worked for some reason, but it doesn't work now. No clue if it's got something to do with maps, addons or some lua runtime flukes I couldn't know how to describe or spot. I reduced gmod to its bare state to test. Here's a video and the test outfit, whatever that's worth...

https://user-images.githubusercontent.com/3782245/158529702-89d93387-d0fe-472f-bd97-e1c1de590832.mp4

[pose_parameter_proxy.txt](https://github.com/CapsAdmin/pac3/files/8259429/pose_parameter_proxy.txt)

**Comment by thegrb93** [comment]
Sorry proxies are a bit of a mystery to me still. Could you explain what they're doing, what they're supposed to be doing?

From what I can gather, you're trying to take the pose_param of a player something and apply it to a bone part?

What's the `-20 + 60*param`? Are these just magic constants you saw worked well?

**Comment by thegrb93** [comment]
I'm also a bit confused why the expression is `"nil,-20 + 60*pose_parameter(\"head_pitch\"),-20 + 40*pose_parameter(\"head_yaw\")"` Shouldn't it be `"-20 + 60*pose_parameter(\"head_pitch\"),-20 + 40*pose_parameter(\"head_yaw\"),nil"`?

**Comment by thegrb93** [comment]
I believe one problem is that you're trying to use `pitch and yaw` as `yaw and roll`. Unfortunately this won't behave well at all since euler angles are so particular.

You might have better luck if you apply these normally to the bone, `pitch and yaw` to `pitch and yaw`, and provide an offset angle.

As for pose_parameter returning 0, I can take a look in the code and see if there's any weirdness. If not, then it might be a gmod issue.

**Comment by thegrb93** [comment]
Looking at the code, it seems the only possibility of it return 0 is if either the name wasn't being passed to the function (this shouldn't be it). Or the 'owner' wasn't the player for some reason.

https://github.com/CapsAdmin/pac3/blob/4b553d98e449d19290c4a7a7ecb2d70d0578f6da/lua/pac3/core/client/parts/proxy.lua#L508-L514

**Comment by Yagira** [comment]
The example file works perfectly fine for me (just had to add a - since it was reversed). https://i.imgur.com/RzNK3tu.mp4

So maybe theres something else messing with it. Are you on the latest version?

**Comment by thegrb93** [comment]
If you can get it to fail again, you can add `print(owner)` after line 510 to see if the player is wrong.

**Comment by pingu7867** [comment]
I'm the fool! I reinstalled pac, now it actually gets the numbers.

I looked back at the fresh download to compare it to the running addon. There were differences. I remember, I guess it mushed up some files when I was switching out the old for the new version or vice versa. It shouldn't've overwritten anything (I think), because I specifically named the older version's folder root "pac-develop-old", but Windows somehow thought they were the same folder and decided to start copying stuff over each other as I frantically cancelled the transfer when it asked to overwrite files.
